protocol/x/vault/keeper/orders_test.go

```diff
@@ -249,79 +249,167 @@ func TestGetVaultClobOrders(t *testing.T) {
+			expectedOrderQuantums: []uint64{
+				20_000_000,
+				20_000_000,
+				20_000_000,
+				20_000_000,
 			},
 		},
 		"Success - Get orders from Vault for Clob Pair 1": {
-			vaultId:  constants.Vault_Clob_1,
-			clobPair: constants.ClobPair_Eth,
-			marketParam: pricestypes.MarketParam{
-				Id:                 constants.TestMarketParams[1].Id,
-				Pair:               constants.TestMarketParams[1].Pair,
-				Exponent:           constants.TestMarketParams[1].Exponent,
-				MinExchanges:       constants.TestMarketParams[1].MinExchanges,
-				MinPriceChangePpm:  4_200, // Set a high min price change to test spread calculation.
-				ExchangeConfigJson: constants.TestMarketParams[1].ExchangeConfigJson,
+			vaultParams: vaulttypes.Params{
+				Layers:                 3,       // 3 layers
+				SpreadMinPpm:           3_000,   // 30 bps
+				SpreadBufferPpm:        4_500,   // 15 bps
+				SkewFactorPpm:          700_000, // 0.7
+				OrderSizePpm:           50_000,  // 5%
+				OrderExpirationSeconds: 4,       // 4 seconds
 			},
-			marketPrice: constants.TestMarketPrices[1],
-			perpetual:   constants.EthUsd_20PercentInitial_10PercentMaintenance,
+			vaultId:            constants.Vault_Clob_1,
+			vaultAssetQuantums: big.NewInt(2_000_000_000), // 2,000 USDC
+			vaultInventory:     big.NewInt(-1_000),
+			clobPair:           constants.ClobPair_Eth,
+			marketParam:        constants.TestMarketParams[1],
+			marketPrice:        constants.TestMarketPrices[1],
+			perpetual:          constants.EthUsd_0DefaultFunding_9AtomicResolution,
+			// To calculate order subticks:
+			// 1. spread = max(spread_min, spread_buffer + min_price_change)
+			// 2. leverage = open_notional / equity
+			// 3. leverage_i = leverage +/- i * order_size_pct (- for ask and + for bid)
+			// 4. skew_i = -leverage_i * spread * skew_factor
+			// 5. a_i = oracle_price * (1 + skew_i) * (1 + spread)^{i+1}
+			//    b_i = oracle_price * (1 + skew_i) / (1 + spread)^{i+1}
+			// 6. subticks needs to be a multiple of subticks_per_tick (round up for asks, round down for bids)
+			// To calculate size of each order
+			// 1. `order_size_ppm * equity / oracle_price`.
 			expectedOrderSubticks: []uint64{
-				// spreadPpm = max(3_000, 1_500 + 4_200) = 5_700
-				// priceSubticks = 3_000_000_000 * 10^(-6 - (-9) + (-9) - (-6)) = 3 * 10^9
-				// a_1 = 3 * 10^9 * (1 + 0.0057)^1 = 3_017_100_000
-				3_017_100_000,
-				// b_1 = 3 * 10^9 * (1 - 0.0057)^1 = 2_982_900_000
-				2_982_900_000,
-				// a_2 = 3 * 10^9 * (1 + 0.0057)^2 = 3_034_297_470
-				// round up to nearest multiple of subticksPerTick=1000.
-				3_034_298_000,
-				// b_2 = 3 * 10^9 * (1 - 0.0057)^2 = 2_965_897_470
-				// round down to nearest multiple of subticksPerTick=1000.
-				2_965_897_000,
+				// spreadPpm = max(3_000, 4_500 + 50) = 4_550
+				// spread = 0.00455
+				// open_notional = -1_000 * 10^-9 * 3_000 * 10^6 = -3_000
+				// leverage = -3_000 / (2_000_000_000 - 3_000) = -3/1_999_997
```

Comment:
same question as below, why are we subtracting from the equity in the denominator?

Reply:
see above